pkg/discovery/kv/kv.go

```diff
@@ -63,9 +63,23 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 	s.ttl = ttl
 	s.path = path.Join(s.prefix, discoveryPath)
 
+	var config *store.Config
```

Comment:
I think I've got an idea on how I can address @NathanMcCauley's suggestion - I'll update the PR shortly once I have it implemented and tested.